misk-action-scopes/src/main/kotlin/misk/scope/coroutine/ActionScopedCoroutineBuilder.kt

```diff
@@ -0,0 +1,33 @@
+package misk.scope.coroutine
+
+import kotlinx.coroutines.CoroutineScope
+import misk.scope.ActionScope
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Wraps a [kotlinx.coroutines.runBlocking] to propagate the current action scope.
+ */
+@Singleton
```

Comment:
it seems like usage of this is
```kotlin
class MyClass @Inject constructor(val actionScopedCoroutineBuilder: ActionScopedCoroutineBuilder) {
  fun doSomething() {
    actionScopedCoroutineBuilder.runBlocking {
      // do async things
    }
  }
}

```
is it worth adding this layer of indirection? seems like we get the same by creating some extension methods?
```kotlin
class MyClass @Inject constructor(val actionScope: ActionScope) {
  fun doSomething() {
    actionScope.runBlocking {
      // do async things
    }
  }
}

```

Reply:
oh I like that!

Reply:
+1 I would prefer to extend `actionScope`

Reply:
modified to just add on `actionScope`